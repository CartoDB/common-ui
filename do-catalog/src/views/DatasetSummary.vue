<template>
  <div class="grid u-flex__justify--center">
    <div class="grid-cell grid-cell--col7 main-column">
      <div class="map-header">
        <img class="" src="../assets/map-samples.jpg" alt="map-sample">
      </div>
      <p class="text is-caption is-txtNavyBlue u-mt--32">
        {{dataset.description}}
      </p>

      <div class="key-variables u-mt--32">
        <h5 class="title is-caption is-txtNavyBlue">Key variables <a href="#" class="is-small">(View all)</a></h5>
        <ul class="grid text is-caption bullet-list u-mt--24">
          <li class="grid-cell grid-cell--col6">Households by age of reference person</li>
          <li class="grid-cell grid-cell--col6">Households by number of dependant children</li>
          <li class="grid-cell grid-cell--col6">Households by dwelling type</li>
          <li class="grid-cell grid-cell--col6">Households by number of rooms</li>
          <li class="grid-cell grid-cell--col6">Households by economic activity</li>
          <li class="grid-cell grid-cell--col6">Households by size</li>
          <li class="grid-cell grid-cell--col6">Households by ethnic groups</li>
          <li class="grid-cell grid-cell--col6">Households by tenure</li>
        </ul>
      </div>
    </div>
    <div class="grid-cell--col1"></div>
    <div class="grid-cell grid-cell--col2 sidebar">
      <ul class="side-characteristics">
        <li class="u-mb--32">
          <h4 class="text is-small is-txtSoftGrey u-mb--10">Licence</h4>
          <p class="text is-caption is-txtNavyBlue">{{dataset_privacy}}</p>
        </li>
        <li class="u-mb--32">
          <h4 class="text is-small is-txtSoftGrey u-mb--10">Country</h4>
          <p class="text is-caption is-txtNavyBlue">{{dataset.country_name}}</p>
        </li>
        <li class="u-mb--32">
          <h4 class="text is-small is-txtSoftGrey u-mb--10">Source</h4>
          <p class="text is-caption is-txtNavyBlue">{{dataset.provider_name}}</p>
        </li>
        <li class="u-mb--32">
          <h4 class="text is-small is-txtSoftGrey u-mb--10">Geography</h4>
          <p class="text is-caption is-txtNavyBlue">{{dataset.geography_name}}</p>
        </li>
        <li class="u-mb--32">
          <h4 class="text is-small is-txtSoftGrey u-mb--10">Temporal aggregation</h4>
          <p class="text is-caption is-txtNavyBlue">{{temporal_aggregation}}</p>
        </li>
        <li class="u-mb--32">
          <h4 class="text is-small is-txtSoftGrey u-mb--10">Update Frequency</h4>
          <p class="text is-caption is-txtNavyBlue">{{update_frequency}}</p>
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex';
import { temporalAggregationName } from '../utils/temporal-agregation-name';

export default {
  name: "DatasetSummary",
  computed: {
    ...mapState({
      dataset: state => state.catalog.dataset
    }),
    temporal_aggregation () {
      return temporalAggregationName(this.dataset.temporal_aggregation)
    },
    update_frequency () {
      return this.dataset.update_frequency ? this.dataset.update_frequency : 'None';
    },
    dataset_privacy () {
      return this.dataset.is_public_data ? 'Public Data' : 'Premium Data';
    }
  },
}
</script>


<style scoped lang="scss">
.map-header {
  overflow: hidden;
  border-radius: 4px;

  img {
    max-width: 100%;
  }
}

.bullet-list {
  li {
    margin-bottom: 12px;
    list-style: inside;
  }
}
</style>