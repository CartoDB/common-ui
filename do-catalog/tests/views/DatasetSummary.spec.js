import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import DatasetSummary from '@/views/DatasetSummary';
import dataset from '../fixtures/dataset';
import keyVariables from '../fixtures/key_variables';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('DatasetSummary.vue', () => {
  const store = createStore();

  it('Renders view properly', () => {
    const datasetSummary = shallowMount(DatasetSummary, {
      localVue,
      store
    });
    expect(datasetSummary).toMatchSnapshot();
  });

  describe('computed', () => {
    it('Renders view properly', () => {
      const datasetSummary = shallowMount(DatasetSummary, {
        localVue,
        store
      });
      const aggregation = datasetSummary.vm.temporal_aggregation;
      expect(aggregation).toBe('Yearly');
    });

    it("Transform update frequency to 'None' when the update frequency is null", () => {
      const datasetSummary = shallowMount(DatasetSummary, {
        localVue,
        store
      });
      const updateFrequency = datasetSummary.vm.update_frequency;
      expect(updateFrequency).toBe('None');
    });

    it('Converts update frequency to readable data', () => {
      let altStore = createStore(dataset.dataset_samples[1]);
      const datasetSummary = shallowMount(DatasetSummary, {
        localVue,
        store: altStore
      });
      const updateFrequency = datasetSummary.vm.update_frequency;
      expect(updateFrequency).toBe('Daily');
    });

    it('Returns premium data when is not public data', () => {
      // let altStore = createStore(dataset.dataset_samples[1])
      const datasetSummary = shallowMount(DatasetSummary, {
        localVue,
        store
      });
      const dataPrivacy = datasetSummary.vm.dataset_privacy;
      expect(dataPrivacy).toBe('Premium Data');
    });

    it('Returns public data when is not public data', () => {
      let altStore = createStore(dataset.dataset_samples[1]);
      const datasetSummary = shallowMount(DatasetSummary, {
        localVue,
        store: altStore
      });
      const dataPrivacy = datasetSummary.vm.dataset_privacy;
      expect(dataPrivacy).toBe('Public Data');
    });
  });
});

function createStore(data = dataset.dataset_samples[0]) {
  const keyVariablesSample = keyVariables.key_variable_samples[0];
  const store = new Vuex.Store({
    state: {
      doCatalog: {
        dataset: data,
        keyVariables: keyVariablesSample
      }
    }
  });
  return store;
}
