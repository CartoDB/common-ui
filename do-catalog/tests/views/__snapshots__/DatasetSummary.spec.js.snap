// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetSummary.vue Renders view properly 1`] = `
<div class="grid u-flex__justify--center">
  <div class="grid-cell grid-cell--col7 main-column">
    <div class="map-header"><img src="@/assets/map-samples.jpg" alt="map-sample"></div>
    <p class="text is-caption is-txtMainTextColor u-mt--32">
      Retail Centrality is the ratio between Retail Turnover Index and Retail Spending Index multiplied by 100. It describes the ability of an area to pin the Retail Spending of its population and of other areas' population down to the local retail trade.
    </p>
    <div class="key-variables u-mt--32">
      <h5 class="title is-caption is-txtMainTextColor">
        Key variables <a href="#" class="is-small">(View all)</a></h5>
      <ul class="text is-caption column-list u-mt--24">
        <li><span>Total Population</span></li>
        <li><span>Number of households</span></li>
        <li><span>Population by gender</span></li>
        <li><span>Median age</span></li>
        <li><span>Population by age and gender</span></li>
        <li><span>Population by race</span></li>
        <li><span>Population by age</span></li>
        <li><span>Population by marital status</span></li>
        <li><span>Population by US citizenship</span></li>
        <li><span>Median Income</span></li>
        <li><span>Income per capita</span></li>
        <li><span>Households by income</span></li>
        <li><span>Poverty status</span></li>
        <li><span>Inequality index</span></li>
        <li><span>Housing units</span></li>
        <li><span>Housing units by type of occupancy</span></li>
        <li><span>Dwellings by type</span></li>
        <li><span>Housing units by construction year</span></li>
        <li><span>Households by family type</span></li>
        <li><span>Median rent</span></li>
        <li><span>Housing units by rent burden</span></li>
        <li><span>Families by type</span></li>
        <li><span>Commuting patterns</span></li>
        <li><span>Car ownership</span></li>
        <li><span>Population by eduaction level</span></li>
        <li><span>Occupations</span></li>
        <li><span>Students by grade</span></li>
        <li><span>Population by language at home</span></li>
      </ul>
    </div>
  </div>
  <div class="grid-cell--col1"></div>
  <div class="grid-cell grid-cell--col2 sidebar">
    <ul class="side-characteristics">
      <li class="u-mb--32">
        <h4 class="text is-small is-txtSoftGrey u-mb--10">Licence</h4>
        <p class="text is-caption is-txtMainTextColor">
          Premium Data
        </p>
      </li>
      <li class="u-mb--32">
        <h4 class="text is-small is-txtSoftGrey u-mb--10">Country</h4>
        <p class="text is-caption is-txtMainTextColor">
          United States of America
        </p>
      </li>
      <li class="u-mb--32">
        <h4 class="text is-small is-txtSoftGrey u-mb--10">Source</h4>
        <p class="text is-caption is-txtMainTextColor">
          Michael Bauer International
        </p>
      </li>
      <li class="u-mb--32">
        <h4 class="text is-small is-txtSoftGrey u-mb--10">Geography</h4>
        <p class="text is-caption is-txtMainTextColor">
          USA - PC 5-digit
        </p>
      </li>
      <li class="u-mb--32">
        <h4 class="text is-small is-txtSoftGrey u-mb--10">
          Temporal aggregation
        </h4>
        <p class="text is-caption is-txtMainTextColor">
          Yearly
        </p>
      </li>
      <li class="u-mb--32">
        <h4 class="text is-small is-txtSoftGrey u-mb--10">
          Update Frequency
        </h4>
        <p class="text is-caption is-txtMainTextColor">
          None
        </p>
      </li>
    </ul>
  </div>
</div>
`;
